java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```
@@ -385,10 +397,130 @@ private API getXPAPI(
+   *
+   * @param enumName Unqualified enum constant name (e.g. STALE_FLAG)
+   * @param classSymbol Enum class symbol for the corresponding constant
+   * @param state Used to find enum class in AST
+   */
+  public boolean isEnumConstantMatchingFlagName(
+      String enumName, Symbol.ClassSymbol classSymbol, VisitorState state) {
+
+    // Skip any enums that have no configuration
+    ImmutableCollection<PiranhaEnumRecord> enumRecords =
+        this.config.getEnumRecordsForName(classSymbol.getSimpleName().toString());
+    if (enumRecords.isEmpty()) {
+      return false;
+    }
+
+    // Check cached matches
+    EnumWithClassSymbol enumWithClassSymbol = new EnumWithClassSymbol(enumName, classSymbol);
+    if (enumsMatchingConstructorArgsCache.containsKey(enumWithClassSymbol)) {
+      return enumsMatchingConstructorArgsCache.get(enumWithClassSymbol);
+    }
+
+    // TODO: Find constructor arguments purely through the symbol table; without the need for state
+    //       Then, we will be able remove the following exception
+    ClassTree enumClassTree = ASTHelpers.findClass(classSymbol, state);
+    if (enumClassTree == null) {
+      throw new PiranhaRuntimeException(
+          "Detected enum constant of class "
+              + classSymbol.className()
+              + ", which is mentioned by"
+              + " Piranha's configuration as part of enumProperties. However, enum definition source"
+              + " is not available when looking at this usage (this can happen when cleaning up flags"
+              + " across build targets)."
+              + "\n\nIf you are trying to use Piranha to clean up enum flags based on the string"
+              + " argument to their constructor (e.g. using enumProperties), the current"
+              + " implementation will fail to match flag usages on a separate target as that"
+              + " containing the enum source, resulting in partial clean up.");
```

👍  (There is a chance that we discover that in practice erroring out here is too strict and we should print a warning, but I definitely prefer to start with the most "fail-fast" possible solution until we know for sure how often this case shows up or if there is any better way to handle it! :) )

---

I'm generally in favor of failing fast 😁 Hiding errors tends to cause problems later...